src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ShopService;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
```

It's a good practice to validate constructor parameters, especially for injected dependencies. If a `null` `strategy` is passed, a `NullPointerException` will occur later in the `process` method. It's better to fail fast by adding a null check here and throwing an `IllegalArgumentException`.